azurerm/internal/services/storage/resource_arm_storage_account.go

```diff
@@ -648,13 +650,23 @@ func resourceArmStorageAccountCreate(d *schema.ResourceData, meta interface{}) e
 		Kind: storage.Kind(accountKind),
 		AccountPropertiesCreateParameters: &storage.AccountPropertiesCreateParameters{
 			EnableHTTPSTrafficOnly: &enableHTTPSTrafficOnly,
-			MinimumTLSVersion:      storage.MinimumTLSVersion(minimumTLSVersion),
 			NetworkRuleSet:         expandStorageAccountNetworkRules(d),
 			IsHnsEnabled:           &isHnsEnabled,
-			AllowBlobPublicAccess:  &allowBlobPublicAccess,
 		},
 	}
 
+	// For US Government Cloud, don't specify "allow_blob_public_access" and "min_tls_version" in request body.
+	// https://github.com/terraform-providers/terraform-provider-azurerm/issues/7812
+	// https://github.com/terraform-providers/terraform-provider-azurerm/issues/8083
+	if env.Name == autorestAzure.USGovernmentCloud.Name {
+		if allowBlobPublicAccess || minimumTLSVersion != string(storage.TLS10) {
+			return fmt.Errorf(`"allow_blob_public_access" and "min_tls_version" are not supported for a Storage Account located in %q`, env.Name)
+		}
+	} else {
```

Comment:
I don't think the logic here is quite right:

1. We don't want to return an error when the fields are passed in the USGovernmentCloud or in the ChinaCloud. If we do the customers will not be able to use the resource which defeats the purpose of these fixes.

2. The if statement is only checking name for autorestAzure.USGovernmentCloud.Name, isn't the TLS issue only in the ChinaCloud? Either way we should be checking for both clouds shouldn't we?

Reply:
@WodansSon , thank you for the comments:
Re # 1: I may be with this solution in this PR. If we do not explicitly throw the error to users, then it means in the Azure.Gov, TF swallows and does not respect/send-to-api whatever users' setting on these fields (take the minimumTLSVer for example), then the next time when users do tf-apply w/o any changes in .tf, he/she will see diff because the .tf contains the user's customized setting on this field but the .tfstate file contains the returned value of this field which is an empty.
And the purpose of this fix is to forbid users from setting these two fields that is by design by the service on Azure.Gov
Re # 2, you are correct but we are lack of testing env of Azure.China. To ensure things are absolutely safe, we may wanna only cover things that are tested

Reply:
@mybayern1974 fair enough... ignore # 1